addons/l10n_vn_edi_viettel/models/account_move.py

```diff
@@ -0,0 +1,766 @@
+            'currencyCode': self.currency_id.name,
+            'adjustmentType': '1',  # 1 for original invoice, which is the case during first issuance.
+            'paymentStatus': self.payment_state in {'in_payment', 'paid'},
+            'cusGetInvoiceRight': True,  # Set to true, allowing the customer to see the invoice.
+            'validation': 1,  # Set to 1, SInvoice will validate tax information while processing the invoice.
+        }
+
+        # When invoicing in a foreign currency, we need to provide the rate, or it will default to 1.
+        if self.currency_id.name != 'VND':
+            invoice_data['exchangeRate'] = self.env['res.currency']._get_conversion_rate(
+                from_currency=self.company_currency_id,
+                to_currency=self.currency_id,
+                company=self.company_id,
+                date=self.invoice_date or self.date,
+            )
+
+        adjustment_origin_invoice = None
+        if self.move_type == 'out_refund':  # Credit note are used to adjust an existing invoice
+            adjustment_origin_invoice = self.reversed_entry_id
+        elif self.l10n_vn_edi_replacement_origin:  # 'Reverse and create invoice' is used to issue a replacement invoice
+            adjustment_origin_invoice = self.l10n_vn_edi_replacement_origin
+
+        if adjustment_origin_invoice:
+            invoice_data.update({
+                'adjustmentType': '5' if self.move_type == 'out_refund' else '3',  # Adjustment or replacement
+                'adjustmentInvoiceType': self.l10n_vn_edi_adjustment_type or '',
+                'originalInvoiceId': adjustment_origin_invoice.l10n_vn_edi_invoice_number,
+                'originalInvoiceIssueDate': self._l10n_vn_edi_format_date(adjustment_origin_invoice.l10n_vn_edi_issue_date),
+                'originalTemplateCode': adjustment_origin_invoice.l10n_vn_edi_invoice_symbol.invoice_template_id.name,
+                'additionalReferenceDesc': self.l10n_vn_edi_agreement_document_name,
+                'additionalReferenceDate': self._l10n_vn_edi_format_date(self.l10n_vn_edi_agreement_document_date),
+            })
+
+        json_values['generalInvoiceInfo'] = invoice_data
+
+    def _l10n_vn_edi_add_buyer_information(self, json_values):
+        """ Create and return the buyer information for the current invoice. """
+        self.ensure_one()
+
+        commercial_partner_phone = self.commercial_partner_id.phone and self._l10n_vn_edi_format_phone_number(self.commercial_partner_id.phone)
+        buyer_information = {
+            'buyerName': self.partner_id.name,
+            'buyerLegalName': self.commercial_partner_id.name,
+            'buyerTaxCode': self.commercial_partner_id.vat,
+            'buyerAddressLine': self.partner_id.street,
+            'buyerPhoneNumber': commercial_partner_phone or '',
+            'buyerEmail': self.commercial_partner_id.email,
```

Comment:
```suggestion
            'buyerEmail': self.commercial_partner_id.email or '',
```
Is it not a problem to have `False` values on your JSON file?

Reply:
I didn't notice any issues with it, but nice catch anyway. It never hurts to be safe